aten/src/ATen/cuda/PeerToPeerAccess.cpp

```diff
@@ -38,6 +39,9 @@ bool get_p2p_access(int dev, int dev_to_access) {
               dev_to_access, " is not a device");
   TORCH_INTERNAL_ASSERT(num_devices_ >= 0, "p2p access cache not initialized");
 
+  static bool using_cudaMallocAsync = (CUDACachingAllocator::allocatorBackend() ==
```

Comment:
So, now that allocatorBackend() isn't doing nontrivial parsing, do we really need to static all these calls?

Reply:
It doesn't hurt right? The only reason I can think it might hurt is if the thread-safety of static init imposes some long-lasting overhead for later accesses (which it might, im not sure how it's implemented).

Reply:
It does hurt because you'll generating `__cxa_guard_acquire` code for the initialization sites, bloating binary size

Reply:
Alright, unstaticed https://github.com/pytorch/pytorch/pull/65365/commits/2b1e0b222dcc66dda03468950dfa91a042180831